### ISSUE TITLE
math formulas crashing?

### ISSUE THREAD

**Issue by cosmopol** [itkach/aard2-android#54]
article "Elektronenvolt" in de-20160505 multi volume slobs just showed rectangles in the text where the math formulas were supposed to be. Now, it is working again...

**Comment by francwalter** [comment]
Can not confirm

**Comment by MHBraun** [comment]
confirmed.
It takes approximately half a second on my S5. Hard to see, but initially there are cryptic signs (too fast to see which ones) which are converted into the math symbols immediately.
Seen in the box Physikalische Einheiten on
- Einheitszeichen
- Formelzeichen
- Dimension
- SI-Einheiten

This is normal behaviour as it should be.

I guess on a slower device this time is more visible.

**Comment by francwalter** [comment]
![screenshot_2016-05-29-17-53-31](https://cloud.githubusercontent.com/assets/927269/15634522/7381bcc0-25c6-11e6-9e03-279ad4b4edc6.png)
Indeed! i didnt pay attention, it is visible on my S4 maybe a second then it is the real signs.
Gosh!

**Comment by cosmopol** [comment]
I know the look of the screenshot and I know that it might take longer to display.

I really meant just black empty rectangles (no pre-formulated strings etc.) and it did not go away for minutes (tested on "icarus e653bk illumina carta" (it has a dual-core cpu AFAIK and is one of the fastest e ink androids out there)).

**Comment by cosmopol** [comment]
Someone has to donate an android e ink reader to Mr Tkach for him to test I suppose. :-)

**Comment by itkach** [comment]
> initially there are cryptic signs

That's TeX markup. It's rendered with [MathJax](https://www.mathjax.org/) (a version of it is packaged  into the dictionary itself, based on MathJax 2.5). It takes time to render math, depends on how much math is there, how complex the formulas are and how fast/slow the device is.

> article "Elektronenvolt" in de-20160505 multi volume slobs just showed rectangles in the text where the math formulas were supposed to be. Now, it is working again

So is it working or not? Just this article or any article with math? 

It would be interesting to check if this device's web browser can render samples at https://www.mathjax.org/ (select SVG rendering mode, that's what Aard2 uses), even though the site runs a newer mathjax version.

> Someone has to donate an android e ink reader to Mr Tkach for him to test I suppose. :-)

Indeed, I can't debug device-specific problems without the device. Or better yet, figure it our yourself and contribute a patch :)

**Comment by cosmopol** [comment]
tried
https://www.mathjax.org/ -> view samples -> SVG
with
latest Google Chrome
and it works.

Open Android is out there. There are at least a few hundred thousand devices in circulation. Reading is done on those.

**Comment by spinxz** [comment]
I can confirm this issue. I am running Aard 2 version 0.34 on a Samsung Galaxy S5 with Android 6.0.1. 
No equations are rendered no matter how long I wait. I just see boxes with the Tex markup.

**Comment by emperormung** [comment]
I have this issue with all of the recent versions on multiple devices (Lollipop, Marshmallow).
Mathematical formulas do not render at all without a network connection). However, in the
same program, formulas DO render in the relevant wikiquote entry. 

Had a fiddle with changing use of Webview.load for styles to evaluateJavascript (in case it was something to do with altered behaviour in more recent Webview versions), but the behaviour is identical (although nothing else broke).

**Comment by itkach** [comment]
@emperormung @MHBraun 

Due to Wikipedia markup changes some recent dictionaries ended up with math as images (loaded from Wikipedia, so requires network). These dictionaries need to be rebuilt with updated mwscrape2slob

See also http://aarddict.org/forum/?place=msg%2Faarddict%2FPKlXbavHvRQ%2Fjy_7dxXhBAAJ

**Comment by MHBraun** [comment]
@itkach 
Thank you for the hint. I updated mwscrape2slob approximately again two weeks ago (after the update end of August). The dewiki-20161212 is compiled with the updated mwscrape2slob. Actually I am scraping the changes for an updated enwiki. This will take another 2 or three months however until all data is updated.
Another option would be to redo the enwiki-20160803 with existing data. Then there would be an interims solution for it. Any need for it?

**Comment by itkach** [comment]
> Another option would be to redo the enwiki-20160803 with existing data. Then there would be an interims solution for it. Any need for it?

@MHBraun Yes, I think so - enwiki-20160803 right now is pretty broken for anyone who is interested in math, it would be nice to have that corrected, even if its with the same data from August, especially if new data is several months away.

**Comment by MHBraun** [comment]
Sorry to reply late on this as I am more focussed on the Google Group. This issues was resolved and does not exist anymore. Igor, you may close the issue.
